### PR TITLE
added Grammar for relative attributes

### PR DESCRIPTION
**Pull request by blackskandix**
just the grammar, so file parsing with these keywords gets possible.
data is not further processed at all

for additional information, sample lines and explanation, please visit: #12

### CODE REVIEW

File: cantools/db/formats/dbc.py [hunk 1]
```diff
@@ -264,6 +264,45 @@ def _create_grammar():
                         - scolon)
     signal_type.setName(SIGNAL_TYPE)
 
+    message_add_sender = Group(Keyword('BO_TX_BU_')
+                               - frame_id
+                               - colon
+                               - Group(delimitedList(node))
+                               - scolon)
+    #message_add_sender.setName('BO_TX_BU_') #not needed?
```
[the remark this answers is not in the document]
Comment (eerimoq) [reply cantools/db/formats/dbc.py]: setName() just makes parser error messages easier to read.

File: cantools/db/formats/dbc.py [hunk 2]
```diff
@@ -264,6 +264,45 @@ def _create_grammar():
...
+
+    attribute_relative = Group(Keyword('BA_REL_')
+                   - QuotedString('"', multiline=True)
+                   - Keyword('BU_SG_REL_')
+                   - word
+                   - Keyword(SIGNAL)
+                   - frame_id
+                   - word
+                   - positive_integer
+                   - scolon)
```
Comment (eerimoq) [inline cantools/db/formats/dbc.py]: Fix indentation to match the rest of the grammar definition.

**Comment by eerimoq** [comment]
Thanks, looks good in general, just a code style comment.

Have a look at the failed tests. Looks like the parser error test case does not pass, which is expected as the exception message now includes the new grammar. I suggest adding setName() and then update the test case.

File: cantools/db/formats/dbc.py [hunk 3]
```diff
@@ -264,6 +264,45 @@ def _create_grammar():
                         - scolon)
     signal_type.setName(SIGNAL_TYPE)
 
+    message_add_sender = Group(Keyword('BO_TX_BU_')
```
Comment (eerimoq) [inline cantools/db/formats/dbc.py]: Use the already declared variable `MESSAGE_TX_NODE`. Apply similar change to other already declared keywords.

**Comment by blackskandix** [comment]
not sure about the tests-part, but to me it seems like the String in tests/test_cantools.py(869):
simply needs an update to:
```
self.assertEqual(
            str(cm.exception),
            "Invalid DBC syntax at line 1, column 1: '>!<abc': Expected "
            "{VERSION | NS_ | BS_ | BU_ | BO_ | CM_ | BA_DEF_ | BA_DEF_DEF_ "
            "| BA_ | VAL_ | VAL_TABLE_ | SIG_VALTYPE_ "
            "| BO_TX_BU_ | BA_DEF_REL_ | BA_DEF_DEF_REL_ | BA_REL_ | EV_}.")
```
to match with the additional keywords. 
But I really don't know anything about your testcases.(?)

**Pull request merged**